File: src/proc_utils.rs
use sysinfo::{ProcessExt, System, SystemExt};

pub struct LeagueProcess {
    pub cmd: Vec<String>,
}

pub fn find_lcu_process() -> Option<LeagueProcess> {
    let mut sys = System::new_all();

    sys.refresh_all();

    for (pid, process) in sys.processes() {
        let name = process.name();
        match name {
            "LeagueClientUx.exe" | "LeagueClientUx" => {
                // DOESN'T WORK!
                // let cmd = process.cmd().to_vec();
                // HAVE TO USE A WHOLE NEW CRATE JUST TO GET CMD!?
                use sysinfo::PidExt;
                let cmd = remoteprocess::Process::new(pid.as_u32())
                    .unwrap()
                    .cmdline()
                    .unwrap()[0]
                    .split("\" \"")
                    .map(|string| String::from(string))
                    .collect::<Vec<String>>();

                let league_process = LeagueProcess {
                    cmd,
                };
                return Some(league_process);
            }
            _ => continue,
        }
    }

    None
}

use std::collections::HashMap;

pub fn parse_cmdline_args(cmdline_args: Vec<String>) -> HashMap<String, String> {
    let mut cmdline_args_parsed = HashMap::new();
    for cmdline_arg in cmdline_args {
        if cmdline_arg.len() > 0 && cmdline_arg.contains("=") {
            let mut split = cmdline_arg[2..].split("=");
            let key = split.next().unwrap().to_owned();
            let value = split.next().unwrap().to_owned();
            cmdline_args_parsed.insert(key, value);
        }
    }
    cmdline_args_parsed
}

File: src/proc_utils.py
""" https://github.com/elliejs/Willump/blob/main/willump/proc_utils.py """

import time
import subprocess
from psutil import process_iter


def get_league_cmdline_args():
    return parse_cmdline_args(get_or_start_league_process())


def parse_cmdline_args(process):
    cmdline_args = process.cmdline()
    cmdline_args_parsed = {}
    for cmdline_arg in cmdline_args:
        if len(cmdline_arg) > 0 and '=' in cmdline_arg:
            key, value = cmdline_arg[2:].split('=')
            cmdline_args_parsed[key] = value
    return cmdline_args_parsed


def get_or_start_league_process():
    process = get_league_process()
    if not process:
        start_league()
        # Wait 5s for league client to start.
        time.sleep(5)
        league_process = get_league_process()

    if not league_process:
        raise FileNotFoundError
    return league_process


def get_league_process():
    for process in process_iter():
        if process.name() in ['LeagueClientUx.exe', 'LeagueClientUx']:
            return process
    return None


def start_league():
    # Start league client.
    league_path = 'C:\\Riot Games\\League of Legends\\LeagueClient.exe'
    subprocess.call([league_path])
